src/Sunburst.ts

```diff
@@ -327,53 +322,325 @@ export class Sunburst implements IVisual {
         }
     }
 
-    public enumerateObjectInstances(options: EnumerateVisualObjectInstancesOptions): VisualObjectInstanceEnumeration {
-        const instanceEnumeration: VisualObjectInstanceEnumeration = SunburstSettings.enumerateObjectInstances(
-            this.settings || SunburstSettings.getDefault(),
-            options
-        );
-
-        if (options.objectName === Sunburst.LegendPropertyIdentifier.objectName) {
-            const topCategories: SunburstDataPoint[] = this.data.root.children;
-            this.enumerateColors(topCategories, instanceEnumeration);
+    public getFormattingModel(): powerbi.visuals.FormattingModel {
+        const visualDatacard: powerbi.visuals.FormattingCard = {
+            description: "Settings for the layout of the visual",
+            displayName: "Layout Settings",
+            uid: "visual_datacard_uid",
+            groups: []
         }
 
-        return (<VisualObjectInstanceEnumerationObject>instanceEnumeration).instances || [];
+        const groupGroup: powerbi.visuals.FormattingGroup = this.getGroupGroup();
+        visualDatacard.groups.push(groupGroup);
+
+        const tooltipGroup: powerbi.visuals.FormattingGroup = this.getTooltipGroup();
+        visualDatacard.groups.push(tooltipGroup);
+
+        const legendGroup: powerbi.visuals.FormattingGroup = this.getLegendGroup();
+        visualDatacard.groups.push(legendGroup);
+
+        return { cards: [visualDatacard] };
+    }
+
+    // eslint-disable-next-line max-lines-per-function
+    private getGroupGroup(): powerbiVisualsApi.visuals.FormattingGroup {
+        const groupGroup: powerbi.visuals.FormattingGroup = {
+            displayName: "Group",
+            uid: "visual_datacard_group_group_uid",
```

Comment:
why do you define uid here?

Reply:
This way of defining the formatting model requires a uid: https://learn.microsoft.com/en-us/power-bi/developer/visuals/format-pane

Reply:
use formattingSettingsService for this

Reply:
Ok, I'll add the formatting service as per the other review